### ISSUE TITLE
Configurable delay for automatic unsuspension

### ISSUE THREAD

**Issue by ethans** [deanoemcke/thegreatsuspender#252]
I want the unsuspend to be automatically when viewing the tab - but it should wait a couple of seconds before doing so (maybe make it configurable).

The reason being that if I'm just viewing all the tabs to see the titles - I don't want it to unsuspend immediately. Also, if you stand on a tab and click suspend - it suspends and immediately unsuspends.

**Comment by deanoemcke** [comment]
There actually already is a delay when switching tabs. However, it is currently set to half a second which may be too short for your purposes. The delay is intended to handle situations where a user is ctrl+tabbing through tabs to get to the one they want. Having this delay time configurable has been requested in another issue: #114 

As for the automatic unsuspending of the current tab, that sounds like a bug that needs fixing. Do you mind if I update the title of this issue to reflect that bug, and leave the other open issue to deal with configurable delay times?

**Comment by ethans** [comment]
Sure, no problem.

The half-second is not enough. As I view that tab, I read the title to decide if this is what I'm looking for, and before I move to the next tab - it's being unsuspended.

Thank you

**Comment by deanoemcke** [comment]
Contrary to my comments above, I will leave this issue open as a feature request for a configurable delay time to automatic tab unsuspension.
I have created a new issue for the bug where you cannot suspend the currently focussed tab #274

**Comment by es6Test** [comment]
how do you set up automatic unsuspension?

**Comment by es6Test** [comment]
I don't like the extra click needed once you reach the tab

**Comment by deanoemcke** [comment]
@es6Test Check the option "Automatically unsuspend tab when it is viewed"

**Comment by pgarciacamou** [comment]
I think this is an excellent feature to have. Being able to modify the time it takes to unsuspend (the same way you can for automatically suspending tabs) is definitely a must-have. With so many tabs opened, it usually takes a few seconds to read the title shown by TGS (the great suspender) and know if that is the tab you wanted.

**Comment by liamjohnston** [comment]
@pgarciacamou but if you're taking the time to read the title before deciding whether you want to unsuspend or not, surely it'd be preferable to instantly unsuspend once you've decided you want it to unsuspend? i.e. clicking the tab / hotkey for refresh / hotkey for unsuspend. 

Like if you had it set to 4 seconds, it seems like it'd be annoying if you identified the tab in say 2 seconds, but had to wait an extra 2 seconds for it to unsuspend...

**Comment by pgarciacamou** [comment]
@liamjohnston that is a fair point! Maybe having this would stress me out? Hmmm 🤔. Could that be avoided by having some type of UI progress-timer gradually showing the user the time left until reload? 

You are a fellow FE dev, how do you have your settings currently? In my case, 20s to suspend (I love it suspending quick) and auto unsuspend. These settings work for me 90% of the time, but when switching modus operandum (viz. running out of coffee), auto-unsuspend stresses me out because if I take 501ms to know if that was the website I was looking for, I end up reloading possibly a heavy tab.

My suggestion is; if it is 500ms at the moment, why not allow the user to decide if they feel better with 501ms or 800ms, see what I mean?

**Comment by pgarciacamou** [comment]
Also, am I the only one that has "auto unsuspend" activated AND ends up reloading half of the websites when casually cruising with the shortcut keys?

**Comment by liamjohnston** [comment]
@pgarciacamou [*secret FED handshake*]

I tried auto-unsuspend a long time ago and it wasn't for me. Other than that, I pretty much use the default settings! Though I could have a much shorter suspend timeout without it being an issue probably.
![screen shot 2018-12-17 at 5 04 46 pm](https://user-images.githubusercontent.com/1134713/50065938-f210b880-021d-11e9-910f-7989c88c0996.png)
Each to their own though :D

Basically it boils down to the amount of settings on the screen. It's already grown a lot from its humble beginnings, and we don't want to cram too much in there. I don't think the demand warrants having another setting (unless this request gets a ton of upvotes I guess).

**Comment by pgarciacamou** [comment]
@liamjohnston 

> [secret FED handshake]

I didn't get it at first 😁

> I don't think the demand warrants having another setting (unless this request gets a ton of upvotes I guess).

It was worth mentioning it, ¯\_(ツ)_/¯.

**Comment by deanoemcke** [comment]
I think there's room here to renegotiate the hard-coded delay before unsuspending a tab. The idea is that casually moving through tabs via alt+tab (or control+tab for mac) does not unsuspend all the intermediate tabs, only the one that you finally end up on.

So the ideally time will not have any of these intermediate tabs unsuspend accidentally.

We need to keep in mind however, that some people will click with the mouse directly on the tab they want (or select directly via some other means such as command+number). These people will want that tab to unsuspend as fast as possible, as they know already that it is the tab they want, and any unnecessary delay will just be painful.

**Comment by ossilator** [comment]
for ctrl-tabbing, the solution is (somewhat obviously) tracking the state of the ctrl key and unsuspending only once it is released, like the os' task switchers typically do with alt/meta.

an imo more usable alternative to speculative tabbing is getting a proper overview of tabs; see #560. but that doesn't impact the validity of the issue as such.
